### DISCUSSION TITLE
predict_model for array data

### DISCUSSION THREAD

**Discussion by JavierLopatin** [pycaret/pycaret#4001] [General]
Hi everyone, 

I have been using the library for a while, and I like it a lot so far. So, thanks for the development. 

I mainly use it to train a predict large array or xarray data, so the predict_model function has been a problem for me so far. This is because it only works with pandas data frames, which in this case is very slow, especially for output. 

I have been using the following to force an array-like output instead of the whole data frame: predict_model(model, data=df).prediction_label.values

Does anyone know a solution or a workaround? using direct prediction functions from sklearn, for example, has produced wrong predictions in my case. 

Cheers, Javier

**Comment by celestinoxp** [comment]
Hi, I'm not sure how to help you... the xarray idea could be looked into... can you provide more information? Maybe you could create a pull-request?

**Comment by JavierLopatin** [comment]
Hi, the integration with xarray is quite easy, there's no problem on that front. I guess the solution would be a more numpy-array type of prediction for the library. This would make the inputs and outputs of the models much lighter and faster for my type of application. 

Cheers, Javier

**Reply by celestinoxp** [reply]
Send a pull-request, I'm sure if it's as you say it will be approved and merged!

**Reply by celestinoxp** [reply]
@JavierLopatin ping... can you create a pull-request ?

**Reply by JavierLopatin** [reply]
Hi, I'm sorry I haven't had the time yet to take a real look at the predict_model function before making a pull request. My idea was to ask if you guys already had a solution for this problem. I'll try to make a PR in the future, but if you'd like to tackle this in parallel, please let me know. 
Thanks, JL
